### ISSUE TITLE
I would like to translate Docs for this project in Hindi Language

### ISSUE THREAD

**Issue by w3Abhishek** [Ciphey/Ciphey#474]
Hi Ciphey,

I am Abhishek Verma from India, and my native language is Hindi. I would like to translate your docs in Hindi so, the guys here in India can read and understand your DOCS.

Please let me know, how I can contribute.

**Comment by bee-san** [comment]
Go ahead! :D Translate the README, the wiki cannot be translated :(

**Comment by w3Abhishek** [comment]
> Go ahead! :D Translate the README, the wiki cannot be translated :(

thanks for allowing me to do so. I will submit my PR very soon

**Comment by lordnodd** [comment]
Hey hindi? Indians speak english fine, we don't need a hindi docs, try doing actual work next time and don't slack off. @bee-san I think its not needed we adopted UK english, most of our law proceedings also takes place in english.

**Comment by w3Abhishek** [comment]
Okay, then I won't. Let's see is there anything else I can improve in the code itself. 
Thanks.

**Comment by Albertosaurusrex** [comment]
> Hey hindi? Indians speak english fine, we don't need a hindi docs, try doing actual work next time and don't slack off. @bee-san I think its not needed we adopted UK english, most of our law proceedings also takes place in english.

I think when the "founder" of the project say that @w3Abhishek can go ahead and translate it, it's that. There's absolutely no harm in translating it?

If @w3Abhishek wants to translate it, let them! It's to absolutely no cost to anyone but them, but if they want to take the time, they can do it?

**Comment by w3Abhishek** [comment]
Thanks @Albertosaurusrex 
I am gonna translate and do the job. :)

**Comment by bee-san** [comment]
> Hey hindi? Indians speak english fine, we don't need a hindi docs, try doing actual work next time and don't slack off. @bee-san I think its not needed we adopted UK english, most of our law proceedings also takes place in english.

Here's the deal. **I don't care**. Translate it to Latin for all I care. Of course, we like more contributions to crackers / decoders -- but we don't care. 

It's a win/win as far as I can see.

**They get hacktoberfest merch**

**We get more contributors, which makes the project look more impressive -- especially in job interviews**.

Go ahead and translate it. And if someone else is reading this right now that knows, I don't know, Latin? Hawaiian? Go ahead too. 

If all you can contribute is a translation go ahead :P

**Comment by Cyclic3** [comment]
> Hey hindi? Indians speak english fine, we don't need a hindi docs, try doing actual work next time and don't slack off. @bee-san I think its not needed we adopted UK english, most of our law proceedings also takes place in english.

I think you are missing the reason we accept translations. Because of the English-speaking origins of the Internet, a majority of technical discussion and implementation happens in English (pretty much all programming languages are english-based). As a result, this distribution also comes across into Ciphey, and with free online translation services, the rest can still divine some meaning from english text.

HOWEVER, this doesn't mean that this is comfortable or easy for them! More technical details and poor phrasing on our part can make it difficult to understand even for native speakers. The most common reaction people have to this is to silently give up and move on, which means we have a smaller community working on our code, which means less gets done.

Imagine, for a second, that the library's source code was written in an obscure derivative of python, and imagine that the maintainers all wrote and read this fluently. You would be able to understand the basic control flow, and might even be able to contribute after enough study, but it would be a lot of work and an unenjoyable chore. In fact, this very reason is why we have decided not to write the whole thing in C++, despite the significant performance benefits of doing so.

And all of this is ignoring the largest group of people who read our docs: those who are trying to use it themselves without asking us. Ciphey is primarily used for fun (by people trying to do CTFs, crack codes they find online, etc), and any resistance they encounter will most likely cause them to drop it and move on to something else, decreasing the chance they start to engage and submit an eye-opening bug report, a revolutionary feature request, or just tidy up our messy code.

TL;DR: I appreciate that pretty much everyone who uses Ciphey *can* speak English, but I reject the idea that they would all *prefer* to. If we can make Ciphey a bit easier to use for just one person, then the negligible hassle of a single PR is worth it. If it doesn't then we've wasted a bit of time. Oh well.

**Comment by mitchm0728** [comment]
Unsubscribe

On 05/10/2020 5:15 PM, "Albert" <notifications@github.com> wrote:
>>
>> Hey hindi? Indians speak english fine, we don't need a hindi docs, try
doing actual work next time and don't slack off. @bee-san I think its not
needed we adopted UK english, most of our law proceedings also takes place
in english.
>
> I think when the "founder" of the project say that @w3Abhishek can go
ahead and translate it, it's that. There's absolutely no harm in
translating it?
>
> If @w3Abhishek wants to translate it, let them! It's to absolutely no
cost to anyone but them, but if they want to take the time, they can do it?

**Comment by chuinzer** [comment]
> doing actual work next time and don't slack off. @.bee-san I think it's not needed we adopted UK English, most of our law proceedings also take place in English.

then what's the point of the creators accept different language translations? you should read what @Cyclic3 said above

**Comment by w3Abhishek** [comment]
Should I go for it or not. Very confused

**Comment by Albertosaurusrex** [comment]
@w3Abhishek Go for it.

**Comment by lordnodd** [comment]
> > Hey hindi? Indians speak english fine, we don't need a hindi docs, try doing actual work next time and don't slack off. @bee-san I think its not needed we adopted UK english, most of our law proceedings also takes place in english.
> 
> I think you are missing the reason we accept translations. Because of the English-speaking origins of the Internet, a majority of technical discussion and implementation happens in English (pretty much all programming languages are english-based). As a result, this distribution also comes across into Ciphey, and with free online translation services, the rest can still divine some meaning from english text.
> 
> HOWEVER, this doesn't mean that this is comfortable or easy for them! More technical details and poor phrasing on our part can make it difficult to understand even for native speakers. The most common reaction people have to this is to silently give up and move on, which means we have a smaller community working on our code, which means less gets done.
> 
> Imagine, for a second, that the library's source code was written in an obscure derivative of python, and imagine that the maintainers all wrote and read this fluently. You would be able to understand the basic control flow, and might even be able to contribute after enough study, but it would be a lot of work and an unenjoyable chore. In fact, this very reason is why we have decided not to write the whole thing in C++, despite the significant performance benefits of doing so.
> 
> And all of this is ignoring the largest group of people who read our docs: those who are trying to use it themselves without asking us. Ciphey is primarily used for fun (by people trying to do CTFs, crack codes they find online, etc), and any resistance they encounter will most likely cause them to drop it and move on to something else, decreasing the chance they start to engage and submit an eye-opening bug report, a revolutionary feature request, or just tidy up our messy code.
> 
> TL;DR: I appreciate that pretty much everyone who uses Ciphey _can_ speak English, but I reject the idea that they would all _prefer_ to. If we can make Ciphey a bit easier to use for just one person, then the negligible hassle of a single PR is worth it. If it doesn't then we've wasted a bit of time. Oh well.

cool, I get your point.

**Comment by harens** [comment]
> Here's the deal. I don't care. Translate it to Latin for all I care. Of course, we like more contributions to crackers / decoders -- but we don't care.

> We get more contributors, which makes the project look more impressive -- especially in job interviews.

I know this issue was opened a while ago (and this is slightly off topic), but I don't know how I feel about this viewpoint.

I spent a lot of time making Ciphey installable via MacPorts https://github.com/Ciphey/Ciphey/issues/583 and making it easier for other package maintainers to add it to their package manager by writing up different pitfalls and solutions https://github.com/Ciphey/CipheyCore/issues/17. 

I'd like to think that these contributions, as well as the contributions of others that put their time and effort into helping, have more value than simply bumping the contributor number for job interviews.

Of course, it goes without saying that I really appreciate all the maintainers' time and effort that goes into maintaining this project. Thank you for all of your work ❤️. But I'd hope that the work of other contributors is also held with value, rather than simply saying that you don't care.

**Comment by bee-san** [comment]
> > Here's the deal. I don't care. Translate it to Latin for all I care. Of course, we like more contributions to crackers / decoders -- but we don't care.
> 
> > We get more contributors, which makes the project look more impressive -- especially in job interviews.
> 
> I know this issue was opened a while ago (and this is slightly off topic), but I don't know how I feel about this viewpoint.
> 
> I spent a lot of time making Ciphey installable via MacPorts #583 and making it easier for other package maintainers to add it to their package manager by writing up different pitfalls and solutions [Ciphey/CipheyCore#17](https://github.com/Ciphey/CipheyCore/issues/17).
> 
> I'd like to think that these contributions, as well as the contributions of others that put their time and effort into helping, have more value than simply bumping the contributor number for job interviews.
> 
> Of course, it goes without saying that I really appreciate all the maintainers' time and effort that goes into maintaining this project. Thank you for all of your work ❤️. But I'd hope that the work of other contributors is also held with value, rather than simply saying that you don't care.

Hey! You're right! Sorry, I have autism so sometimes the words come out of me wrong. I meant it like "all contributions are actually great. I don't care if you do a small spelling fix, or a very large feature request. Both are equally amazing and I am so thankful that someone has taken time out of their busy day to do _something_ for this project".

And also, more contributors is better for the project. We learn how our wiki works, and our contribution guidelines. The more contributors we get, the more feedback we get on the experience and the more we can improve this whole project!

Contributions from people that know nothing about how Ciphey works is great, as it means our documentation has worked!

**Comment by harens** [comment]
Thanks for clarifying @bee-san. I'm sorry I misunderstood what you meant, and I'm glad that all contributions are appreciated.

I really did mean it though when I say thank you to you and the other maintainers for putting your time and effort into this project. Your dedication into making a well-documented, super fast and easy-to-use tool for everyone to use is greatly appreciated. Thank you!

> PS: @harens the Ciphey core team are working on our hashing stuff, if you wanted to make any Macports for that (it's 100% Python, promise!) please do. It would mean a lot to us, and would probably make it easier in the future as Search-that-hash is going to be the hashing module used in Ciphey :))

I'd be honoured :) It looks relatively straightforward (much more than CipheyCore!) so I'll definitely have a go at that.
